async_channel/producer.py

```diff
@@ -116,21 +116,18 @@ async def wait_for_processing(self) -> None:
         It will wait until all consumers have notified that their consume() method have ended
         """
         await asyncio.gather(
-            *[consumer.queue.join() for consumer in self.channel.get_consumers()]
+            *(consumer.queue.join() for consumer in self.channel.get_consumers())
         )
 
     async def synchronized_perform_consumers_queue(self, priority_level) -> None:
         """
         Empties the queue synchronously for each consumers
         :param priority_level: the consumer minimal priority level
         """
-        for consumer in [
-            consumer
-            for consumer in self.channel.get_consumers()
-            if consumer.priority_level <= priority_level
-        ]:
-            while not consumer.queue.empty():
-                await consumer.perform(await consumer.queue.get())
+        for consumer in self.channel.get_consumers():
```

Comment:
It's faster for any size of `get_consumers()`?

Reply:
Yes, the main things is to avoid creating intermediary iterators if it will be used only once (difference is very slight though)
https://github.com/Drakkar-Software/OctoBot-Benchmarks/pull/21